### PR TITLE
user-story/1442-as-a-user,-i-want-to-track-detections-over-multiple-videos

### PR DESCRIPTION
**Pull request by briemla**
OP#1442

### CODE REVIEW

File: OTVision/dataformat.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+METADATA: str = "metadata"
```
Comment (briemla) [inline OTVision/dataformat.py:1]: This file will be extended in OP#1541 New output format. All files will be checked and strings will be replaced in OP#1541

File: OTVision/helpers/files.py [hunk 2]
```diff
@@ -282,13 +285,50 @@ def denormalize_bbox(
...
-        log.debug("Dict denormalized")
-    else:
-        log.debug("Dict was already denormalized")
+    log.debug("Denormalize frame wise")
+    otdict = _denormalize_transformation(otdict, keys_width, keys_height, metadata)
+    return otdict
+
+
+# TODO: Type hint nested dict during refactoring
+def _denormalize_transformation(
```
Comment (briemla) [inline OTVision/helpers/files.py:294]: Denormalization and normalization have been separated again. The update of metadata is different in both methods. Therefore, the separation was necessary.

Metadata will now be read by frame, as frames belong to different video files.

File: OTVision/helpers/files.py [hunk 3]
```diff
@@ -315,48 +357,57 @@ def normalize_bbox(
...
         log.debug("Dict normalized")
     else:
         log.debug("Dict was already normalized")
     return otdict
 
 
 # TODO: Type hint nested dict during refactoring
-def _normal_transformation(
-    otdict: dict, direction: str, keys_width: list[str], keys_height: list[str]
+def _normalize_transformation(
```
Comment (briemla) [inline OTVision/helpers/files.py:373]: See comment at `_denormalize_transformation`

File: OTVision/track/preprocess.py [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+    recording will be merged together. The time gape to separate two recordings from
+    each other is defined by `self.time_without_frames`.
+
+    Returns:
+        Preprocess: preprocessor for tracking
+    """
+
+    time_without_frames: timedelta
+
+    def __init__(self, no_frames_for: timedelta = timedelta(minutes=1)) -> None:
```
Comment (randy-seng) [inline OTVision/track/preprocess.py:213]: Can we rename `no_frames_for` to `time_without_frames`?

**Review by randy-seng — CHANGES_REQUESTED** [review]
Looks good! I just have a few comments to be resolved~

File: OTVision/detect/detect.py [hunk 5]
```diff
@@ -19,18 +19,28 @@
...
 from OTVision.helpers.files import get_files, write_json
 from OTVision.helpers.log import log, reset_debug, set_debug
+from OTVision.track.preprocess import DATE_FORMAT, OCCURRENCE
 
 from . import yolo
 
+FILE_NAME_PATTERN = (
+    "(?P<prefix>[A-Za-z0-9]+)"
+    "_FR(?P<frame_rate>\\d+)"
+    "_(?P<start_date>\\d{4}-\\d{2}-\\d{2}_\\d{2}-\\d{2}-\\d{2})\\..*"
```
[the remark this answers is not in the document]
Comment (briemla) [reply OTVision/detect/detect.py:41]: Thanks, nice feature.

**Review by randy-seng — APPROVED** [review]

**Review by martinbaerwolff — APPROVED** [review]
Looks good, I ran detection and tracking using your branch and the IDs are consecutive for consecutive .ottrk files from consecutive videos :)

However, I got a log `metadata updated from historic format to new format` during tracking, but I assume this can be fixed with OP#1541

**Pull request merged**